### PR TITLE
lsof: 4.98.0 -> 4.99.0

### PR DESCRIPTION
**Pull request by trofi**
While at it dropped local patch as upstream does not retain hostname and build user by default.

Changes: https://github.com/lsof-org/lsof/releases/tag/4.99.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/lsof/default.nix [hunk 1]
```diff
@@ -37,6 +35,10 @@ stdenv.mkDerivation rec {
     for filepath in $(find dialects/${dialect} -type f); do
       sed -i "s,/usr/include,$LSOF_INCLUDE,g" $filepath
     done
+
+    # Wipe out development-only flags from CFLAGS embedding
+    make version.h
+    sed -e "s|$NIX_STORE/[a-z0-9]\{32\}-|$NIX_STORE/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-|g" -i version.h
```
Comment (bjornfor) [inline pkgs/development/tools/misc/lsof/default.nix:41]: Use nukeReferences / nuke-refs instead of sed?

Comment (trofi) [reply pkgs/development/tools/misc/lsof/default.nix]: Good idea! I also noticed that initial version did not work as `version.h` gets rebuilt on each `make` run. Added a workaround for it to remove phony depend as `sed -i lib/dialects/*/Makefile -e 's/version.h:\s*FRC/version.h:/'`.

File: pkgs/development/tools/misc/lsof/default.nix [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
   };
 
-  patches = [
-    ./no-build-info.patch
-  ];
-
-  postPatch = lib.optionalString stdenv.hostPlatform.isMusl ''
+  postPatch = ''
+    patchShebangs --build lib/dialects/*/Mksrc
+    # Do not re-build version.h in every 'make' to alow nuke-refs below.
```
[the remark this answers is not in the document]
Comment (trofi) [reply pkgs/development/tools/misc/lsof/default.nix]: Fixed typo. Thank you!

**Comment by amjoseph-nixpkgs** [comment]
> While at it dropped local patch as upstream does not retain hostname and build user by default.

Would you mind splitting that out into a separate commit?  It will make it easier if people on exotic platforms need to revert/bisect around it.  Both commits can stay in the same PR.  Thanks!

**Comment by bjornfor** [comment]
>> While at it dropped local patch as upstream does not retain hostname and build user by default.
>
> Would you mind splitting that out into a separate commit?

I assumed it was only possible to remove the patch after the update? If it can be removed before, then I agree with splitting the commit.

**Pull request closed without merging**